### PR TITLE
LPS-186915: Automatically (un)publishing API Application based on applicationStatus

### PR DESCRIPTION
**Pull request by 4lejandrito**
**What is new?**
- Update `ObjectEntryModelListener` because we need the `onAfterUpdate` method and if after updating an Object Entry that has the same ERC doesn't change any value.
- Add `onAfterCreate` and `onAfterUpdate` to call the APIApplicationPublisher
- Add some tests cases

**Note**: This PR doesn't follow any changes from this [PR](https://github.com/liferay-headless/liferay-portal/pull/1317), so there will be some errors. However, I will create this PR and rebase until the previous one is been merged.

**How to reproduce it?**
- Deploy `object-api` and `object-service`
- Deploy `headless-builder-api` and `headless-builder-impl`
- Enable `feature.flag.LPS-186757=true`

**JIRA Related Tickets**
https://liferay.atlassian.net/browse/LPS-186915

### CODE REVIEW

**Comment by 4lejandrito** [comment]
ci:reopen

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/HeadlessBuilderObjectEntryModelListener.java
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/HeadlessBuilderObjectEntryModelListener.java:1]: @ccorrea

**Comment by brianchandotcom** [comment]
Merged. Thank you.
View total diff: https://github.com/brianchandotcom/liferay-portal/compare/601467d85d...b9800fcd30

**Comment by brianchandotcom** [comment]
@sergiojimcos @4lejandrito see changes in my origin. Thx.

**Pull request closed without merging**